document/core/exec/instructions.rst

```diff
@@ -1248,24 +1262,32 @@ Reference Instructions
      \\ \quad S;
        \begin{array}[t]{@{}l@{}}
        (\REFARRAYADDR~a_1)~(\I32.\CONST~d) \\
-       (\REFARRAYADDR~a_2)~(\I32.\CONST~s)~(\ARRAYGET~y) \\
+       (\REFARRAYADDR~a_2)~(\I32.\CONST~s)~\X{get} \\
        (\ARRAYSET~x) \\
        (\REFARRAYADDR~a_1)~(\I32.\CONST~d+1)~(\REFARRAYADDR~a_2)~(\I32.\CONST~s+1)~(\I32.\CONST~n)~(\ARRAYCOPY~x~y) \\
        \end{array}
      \\ \qquad
-     (\otherwise, \iff d \leq s)
+     \begin{array}[t]{@{}r@{~}l@{}}
+     (\otherwise, \iff & d \leq s \\
+                  \land & (\X{get} = \ARRAYGET~y \vee \X{get} = \ARRAYGETU~y) \\
+                  \land & (F.\AMODULE.\MITYPES[x] = \TARRAY~\mut~\packedtype \Longleftrightarrow \X{get} = \ARRAYGETU~y))
```

Comment:
Can you think of a better way to express this? Is there a better symbol to use for bidirectional implication than `\Longleftrightarrow`?

Reply:
The symbol would be correct. But I would suggest to factor out the choice of `get` into an auxiliary function which does a case distinction on the storage type, i.e., something like
```
getfield(valtype) = array.get
getfield(packedtype) = array.get_u
```

Reply:
Thanks, I've written this up now. PTAL!